### PR TITLE
OCPBUGS-3441: Update cluster-authentication-operator not to go degraded without console [4.13] 

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

File: pkg/controllers/configobservation/configobservercontroller/observe_config_controller.go [hunk 1]
```diff
@@ -79,21 +81,30 @@ func NewConfigObserver(
...
+		ClusterVersionLister: configInformer.Config().V1().ClusterVersions().Lister(),
+		InfrastructureLister: configInformer.Config().V1().Infrastructures().Lister(),
+		OAuthLister_:         configInformer.Config().V1().OAuths().Lister(),
+		ResourceSync:         resourceSyncer,
+		PreRunCachesSynced:   preRunCacheSynced,
+	}
+
+	// Check if the Console capability is enabled on the cluster and sync and add its informer and lister.
+	if enabledClusterCapabilities.Has("Console") {
+		preRunCacheSynced = append(preRunCacheSynced, configInformer.Config().V1().Consoles().Informer().HasSynced)
```
Comment (stlaz) [inline pkg/controllers/configobservation/configobservercontroller/observe_config_controller.go:99]: I think this should be `listers.PreRunCachesSynced = append(listers.PreRunCachesSynced, ...` to make sure the above struct contains the right slice

**Review by stlaz — CHANGES_REQUESTED** [review]

**Comment by jhadvig** [comment]
🥳

**Pull request merged**